File: leetcode3/최원준/3838. Weighted Word Mapping.py
#

'''
1. 아이디어 :
-

2. 시간복잡도 :
    O(n * m) n = words.length , m =words[i].length

3. 자료구조/알고리즘 :
-

'''
class Solution:
    def mapWordWeights(self, words: List[str], weights: List[int]) -> str:
        return "".join(chr(ord('z') - sum(weights[ord(c) - ord('a')] for c in word) %26 ) for word in words)

File: leetcode3/염혜정/3838. Weighted Word Mapping.java
// 97 ~ 122

class Solution {
    public String mapWordWeights(String[] words, int[] weights) {
        StringBuilder sb = new StringBuilder();
        for (String word : words) {
            int sum = 0;
            for (char c : word.toCharArray()) {
                int num = (int) c - 97;
                sum += weights[num];
            }
            int mod = sum % 26;
            sb.append((char)('z' - mod));
        }
        return sb.toString();
    }
}
